### PR TITLE
Feature: add raven gateway CRD

### PR DESCRIPTION
**Pull request by njucjc**
kind feature
> /kind question
> /kind design
> /sig ai
> /sig iot
> 
sig network
> /sig storage

#### What this PR does / why we need it:
migrate `raven-controller-manager` to `yurt-manager`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -135,3 +136,11 @@ replace (
...
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.22
 )
+
+exclude (
+	k8s.io/api v0.23.2
+	k8s.io/apimachinery v0.23.2
+	k8s.io/apiserver v0.23.2
+	k8s.io/client-go v0.23.2
+	k8s.io/component-base v0.23.2
+)
```
[the remark this answers is not in the document]
Comment (njucjc) [reply go.mod:146]: this project (v0.22.3) is conflict with github.com/projectcalico/api (v0.23.2)

**Review by luckymrwang — APPROVED** [review]

File: pkg/webhook/gateway/v1alpha1/gateway_handler.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		util.GenerateValidatePath(gvk),
+		ctrl.NewWebhookManagedBy(mgr).
+			For(&v1alpha1.Gateway{}).
+			WithDefaulter(webhook).
+			WithValidator(webhook).
+			Complete()
+}
+
+// +kubebuilder:webhook:path=/validate-raven-openyurt-io-gateway,mutating=false,failurePolicy=fail,sideEffects=None,admissionReviewVersions=v1;v1beta1,groups=raven.openyurt.io,resources=gateways,verbs=create;update,versions=v1alpha1,name=validate.raven.v1alpha1.gateway.openyurt.io
+// +kubebuilder:webhook:path=/mutate-raven-openyurt-io-gateway,mutating=true,failurePolicy=fail,sideEffects=None,admissionReviewVersions=v1;v1beta1,groups=raven.openyurt.io,resources=gateways,verbs=create;update,versions=v1alpha1,name=mutate.raven.v1alpha1.gateway.openyurt.io
```
Comment (rambohe-ch) [inline pkg/webhook/gateway/v1alpha1/gateway_handler.go:48]: It seems that raven mutate webhook  doesn't exist, so how about remove this settings?

Comment (njucjc) [reply pkg/webhook/gateway/v1alpha1/gateway_handler.go:48]: https://github.com/njucjc/openyurt/blob/1cfa1530f95925ee061756dc304378734fab1028/pkg/apis/raven/v1alpha1/default.go#L26

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**